Task_2/src/components/Cart/Cart.tsx

```diff
@@ -1,11 +1,62 @@
 import React from "react";
+import { useState, useEffect, useCallback, useContext } from "react";
+import CartProductItem from "./CartProductItem";
+import ProductModel from "../../Models/ProductModel";
+import BillComponent from "./BillComponent";
+import CartContext from "../../Store/ShoppingCart";
+import { DiJava } from "react-icons/di";
+import ProductList from "../Products/ProductList";
 
 const CartPage: React.FC = () => {
+  const cartCtx = useContext(CartContext);
+
+  const [data, setData] = useState<ProductModel[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const fetchData = useCallback(async () => {
+    try {
+      setIsLoading(true);
+      const ProductList: ProductModel[] = [];
+      for (let i of cartCtx?.items ?? []) {
```

Comment:
Why we are calling api's inside a loop ? even if it's needed then we should have parallel API calling at place. First of all it should not be in loop. Else use promise.all() for parallel API calls @codal-uchoudhary

Reply:
done